cli/src/config.rs

```diff
@@ -229,12 +230,36 @@ impl UnresolvedConfigEnv {
             config_dir.push("conf.d");
             ConfigPath::new(config_dir)
         });
+
         use ConfigPath::*;
-        if let Some(path @ Existing(_)) = home_config_path {
-            paths.push(path);
-        } else if let (Some(path @ New(_)), None) = (home_config_path, &platform_config_path) {
-            paths.push(path);
+
+        match home_config_path {
+            Some(path @ Existing(_)) => paths.push(path),
+            Some(path @ New(_)) if platform_config_path.is_none() => paths.push(path),
+            _ => (),
+        }
```

Comment:
This `match home_config_path` block looks like an unrelated refactoring. Especially since it seems like it gets removed again in the last commit, could you undo the refactoring in the second commit? Sorry if I'm missing something